packages/editor/src/store/selectors.js

```diff
@@ -1708,6 +1733,10 @@ export const getInserterItems = createSelector(
 				return false;
 			}
 
+			if ( isAncestorOf( state, reusableBlock.clientId, rootClientId ) ) {
+				return false;
```

Comment:
From memory, `getInserterItems` has some fairly thorough unit tests. Let's keep that up and add a test case for this new edge case.